src/main/java/ui/GUIController.java

```diff
@@ -102,11 +125,54 @@ public void panelFilterExpressionChanged(FilterPanel changedPanel) {
                 .collect(Collectors.toList());
     }
 
-    private void updateAPIBox(UpdateRateLimitsEvent e) {
-        Platform.runLater(() -> apiBox.setText(String.format("%s/%s",
-                                                             e.remainingRequests,
-                                                             Utility.minutesFromNow(e.nextRefreshInMillisecs)))
-        );
+    /**
+     * Updates UI components using the API rate limits information.
+     * @param e The current API rate limits
```

Comment:
ok right. Changed to "RateLimitsUpdatedEvent object that holds the current API quota information."